### PR TITLE
Hide ErrorOverlay iframe

### LINKED ISSUE #4: Entire application crashes when refactoring exercises
## Versions
- `@kentcdodds/react-workshop-app` version: ^2.12.0
- `node` version: v10.18.1 
- `npm` version: 6.13.4

## What you did:
This happens on occasion but when you write code that isn't valid and React crashes (as expected), the instructions also disappear in the app. 

While you can look at the markdown file that is sitting next to the exercise.js file, it would be a much nicer workflow to have the instructions still present. 

More specifically, in [Exercise 04 in the React Hooks workshop](https://github.com/kentcdodds/react-hooks/blob/f9346258c209fd72ec1465df80b2d8982dc1ec5e/src/exercise/04.js#L71-L79), we are instructed to lift state from the `Board` component to the `Game` Component with some suggested JSX that you give us in the instructions. 

A natural first step for me was to paste that JSX into the `Game` component so I can work with the variables that are being passed to the various components.

## What happened:
The whole app crashes! Again, I did expect this but would still like to reference the instructions that were given

Image 2020-07-15 at 6 37 52 PM

## Suggested solution:

I expect that an ErrorBoundary around the `exerciseElement` would do the trick but would like your input on this or if I'm even thinking about this in the right way. 😄 

[Relevant Code:](https://github.com/kentcdodds/react-workshop-app/blob/b21874bb1579e737dce463b3803985ba001459d6/src/react-app.js#L399-L419)

```javascript
<React.Suspense
  fallback={
    <div
      // this centers the loading with the loading
      // for the instructions
      css={[
        totallyCenteredStyles,
        mq({marginBottom: ['auto', 'auto', 80]}),
      ]}
    >
      Loading...
    </div>
  }
>
  <div
    className="exercise-container render-container"
    css={mq({paddingBottom: [0, 0, '2rem']})}
  >
    {exerciseElement}
  </div>
</React.Suspense>
```

### PR DESCRIPTION
**Pull request by zacjones93**
## Changes made
- iframe display none 
  - Hides the ErrorOverlay iframe that create-react-app enables.

**Why**:
This fixes #4, the iframe hides the workshop instructions which often times can help as you refactor an exercise in the extra credit lessons.

**How**:
In [workshop-app-styles.css](https://github.com/kentcdodds/react-workshop-app/blob/master/other/workshop-app-styles.css): 
```
iframe {
  display: none;
}
```

**Checklist**:

- [x] Documentation (inline)
- [ ] Tests N/A
- [x] Ready to be merged
      

I had to do two things to get the example running locally:

1. I had to delete the [react-workshop-app/example/react-fundamentals/src/examples/example.tsx](https://github.com/kentcdodds/react-workshop-app/blob/master/example/react-fundamentals/src/examples/example.tsx) file as the set up would 'detect TypeScript', add a tsConfig file and error out because TypeScript wasn't installed.
Image 2020-07-22 at 4 11 43 PM

2. In [react-fundamentals index page](https://github.com/kentcdodds/react-workshop-app/blob/master/example/react-fundamentals/src/index.js) I had to set `concurrentMode` to `false` because the app would compile with it set to true would produce the error below 
Image 2020-07-22 at 4 13 00 PM

```
codegen`module.exports = require('@kentcdodds/react-workshop-app/codegen')({
  options: {concurrentMode: false},
})`
```

### CODE REVIEW

File: other/workshop-app-styles.css [hunk 1]
```diff
@@ -16,3 +16,13 @@ hr {
...
   margin-bottom: 30px;
 }
+
+/* 
+This hides the Create-React-App ErrorOverlay that will overlay the whole application when React fails to compile. The problem here is that it also overlays the workshop instructions with the example
+
+Reference: https://github.com/kentcdodds/react-workshop-app/issues/4
+
+*/
+iframe {
```
Comment (kentcdodds) [inline other/workshop-app-styles.css:26]: Let's leave the isolated pages out of this. On the isolated pages the error overlay can be kinda handy.

```suggestion
body:not([class*="isolated"]) iframe {
```

**Comment by kentcdodds** [comment]
@all-contributors please add @zacjones93 for code

File: other/workshop-app-styles.css (hunk 1 above)
Comment (zacjones93) [reply other/workshop-app-styles.css]: oh good call

**Pull request merged**